CalibMuon/CSCCalibration/plugins/CSCBadChambersConditions.cc

```diff
@@ -35,9 +33,7 @@ CSCBadChambersConditions::ReturnType
 CSCBadChambersConditions::produceBadChambers(const CSCBadChambersRcd& iRecord)
 {
   //need a new object so to not be deleted at exit
-  CSCBadChambers* mydata=new CSCBadChambers( *cndbBadChambers );
-  return mydata;
-  
+  return std::make_unique<CSCBadChambersConditions::ReturnType::element_type>( *(prefillBadChambers()) );
```

Comment:
This should just be
```C++
return ReturnType( prefillBadChambers() );
```
What you have will cause a memory leak because the object made from prefillBadChambers() gets copied but the original never gets deleted.